lib/pymedphys/_experimental/streamlit/apps/sum_doses.py

```diff
@@ -272,47 +337,3 @@ def sum_doses_in_datasets(
     ds_summed.PixelData = pixel_array_summed.tobytes()
 
     return ds_summed
-
-
-def main():
-
-    left_column, right_column = st.beta_columns(2)
-
-    with left_column:
-        st.write("## Upload DICOM RT Dose files")
-        files = st.file_uploader(
-            "Upload at least two DICOM RT Dose files whose doses you'd "
-            "like to add together. The first file uploaded will be "
-            "used as a template for the summed DICOM RT Dose file.",
-            ["dcm"],
-            accept_multiple_files=True,
-        )
-
-    if st.button("Sum Doses"):
```

Comment:
I'd was aiming to provide the ability to sum _more_ than two doses. By having a button, summing won't triggered (and unused files won't be generated) until the user is ready/has uploaded all files they want to sum.

I don't think there's a way to achieve this without a button..

Reply:
All good, a button's fine 🙂